torch/testing/_internal/common_methods_invocations.py

```diff
@@ -11191,6 +11192,9 @@ def ref_pairwise_distance(input1, input2):
                # RuntimeError: !lhs.isAliasOf(rhs)INTERNAL ASSERT FAILED at
                # "../torch/csrc/jit/passes/utils/check_alias_annotation.cpp":103, please report a bug to PyTorch.
                DecorateInfo(unittest.skip("Skipped!"), 'TestJit', 'test_variant_consistency_jit'),
+               # AssertionError: None mismatch: torch.complex128 is not None
+               DecorateInfo(unittest.expectedFailure, 'TestDtypeCustomRules',
+                            'test_custom_rules', dtypes=(torch.complex64, torch.complex128)),
```

Comment:
Complete Stack Trace:
```
raceback (most recent call last):
  File "/home/kshiteej/Pytorch/pytorch_complex_convolution.py/test/jit/test_dtype_analysis.py", line 361, in test_custom_rules
    self.custom_rules_test_base(device, dtype, op)
  File "/home/kshiteej/Pytorch/pytorch_complex_convolution.py/test/jit/test_dtype_analysis.py", line 357, in custom_rules_test_base
    self.assert_output_dtype_equal(expected_res, graph)
  File "/home/kshiteej/Pytorch/pytorch_complex_convolution.py/test/jit/test_dtype_analysis.py", line 323, in assert_output_dtype_equal
    self.assert_tensor_dtype_equal(expected_res, actual_dtype[0])
  File "/home/kshiteej/Pytorch/pytorch_complex_convolution.py/test/jit/test_dtype_analysis.py", line 332, in assert_tensor_dtype_equal
    self.assertEqual(tensor_output.dtype, graph_dtype)
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 2160, in assertEqual
    assert_equal(
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1084, in assert_equal
    raise error_metas[0].to_error()
AssertionError: None mismatch: torch.complex128 is not None
```

Reply:
what does `TestDtypeCustomRules. test_custom_rules` test? why is this an expected failure?

Reply:
I am not sure. Looking at the test it seems to check if dtype is propagated correctly in JIT traced graph. (I was planning to point someone from the JIT team to unblock this PR)

https://github.com/pytorch/pytorch/blob/ad028e5e09d5ddffee63c49b723594a84b8a90e4/test/jit/test_dtype_analysis.py#L334-L357